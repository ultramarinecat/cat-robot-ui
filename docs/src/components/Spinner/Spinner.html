<!DOCTYPE html>

<html>
<head>
  <title>Spinner.js</title>
  <meta http-equiv="content-type" content="text/html; charset=UTF-8">
  <meta name="viewport" content="width=device-width, target-densitydpi=160dpi, initial-scale=1.0; maximum-scale=1.0; user-scalable=0;">
  <link rel="stylesheet" media="all" href="../../../docco.css" />
</head>
<body>
  <div id="container">
    <div id="background"></div>
    
      <ul id="jump_to">
        <li>
          <a class="large" href="javascript:void(0);">Jump To &hellip;</a>
          <a class="small" href="javascript:void(0);">+</a>
          <div id="jump_wrapper">
          <div id="jump_page_wrapper">
            <div id="jump_page">
              
                
                <a class="source" href="../App/App.html">
                  src/components/App/App.js
                </a>
              
                
                <a class="source" href="../App/ErrorBoundary.html">
                  src/components/App/ErrorBoundary.js
                </a>
              
                
                <a class="source" href="../App/i18n.html">
                  src/components/App/i18n.js
                </a>
              
                
                <a class="source" href="../Auth/Auth.html">
                  src/components/Auth/Auth.js
                </a>
              
                
                <a class="source" href="../Auth/AuthActions.html">
                  src/components/Auth/AuthActions.js
                </a>
              
                
                <a class="source" href="../Auth/AuthReducer.html">
                  src/components/Auth/AuthReducer.js
                </a>
              
                
                <a class="source" href="../Button/Button.html">
                  src/components/Button/Button.js
                </a>
              
                
                <a class="source" href="../Cat/Cat.html">
                  src/components/Cat/Cat.js
                </a>
              
                
                <a class="source" href="../Cat/CatActions.html">
                  src/components/Cat/CatActions.js
                </a>
              
                
                <a class="source" href="../Cat/CatReducer.html">
                  src/components/Cat/CatReducer.js
                </a>
              
                
                <a class="source" href="../Cat/fulltilt.html">
                  src/components/Cat/fulltilt.js
                </a>
              
                
                <a class="source" href="../Connection/Connection.html">
                  src/components/Connection/Connection.js
                </a>
              
                
                <a class="source" href="../Connection/ConnectionActions.html">
                  src/components/Connection/ConnectionActions.js
                </a>
              
                
                <a class="source" href="../Connection/ConnectionReducer.html">
                  src/components/Connection/ConnectionReducer.js
                </a>
              
                
                <a class="source" href="../Connection/messages.html">
                  src/components/Connection/messages.js
                </a>
              
                
                <a class="source" href="../Intro/Intro.html">
                  src/components/Intro/Intro.js
                </a>
              
                
                <a class="source" href="../Intro/IntroActions.html">
                  src/components/Intro/IntroActions.js
                </a>
              
                
                <a class="source" href="../Particles/Particles.html">
                  src/components/Particles/Particles.js
                </a>
              
                
                <a class="source" href="../Pulse/Pulse.html">
                  src/components/Pulse/Pulse.js
                </a>
              
                
                <a class="source" href="../Router/Router.html">
                  src/components/Router/Router.js
                </a>
              
                
                <a class="source" href="../Sound/CatSound.html">
                  src/components/Sound/CatSound.js
                </a>
              
                
                <a class="source" href="../Sound/Mute.html">
                  src/components/Sound/Mute.js
                </a>
              
                
                <a class="source" href="../Sound/Sound.html">
                  src/components/Sound/Sound.js
                </a>
              
                
                <a class="source" href="../Sound/TurningSound.html">
                  src/components/Sound/TurningSound.js
                </a>
              
                
                <a class="source" href="Spinner.html">
                  src/components/Spinner/Spinner.js
                </a>
              
                
                <a class="source" href="../Typing/Typing.html">
                  src/components/Typing/Typing.js
                </a>
              
                
                <a class="source" href="../../configureStore.html">
                  src/configureStore.js
                </a>
              
                
                <a class="source" href="../../index.html">
                  src/index.js
                </a>
              
                
                <a class="source" href="../../serviceWorker.html">
                  src/serviceWorker.js
                </a>
              
                
                <a class="source" href="../../setupTests.html">
                  src/setupTests.js
                </a>
              
            </div>
          </div>
        </li>
      </ul>
    
    <ul class="sections">
        
          <li id="title">
              <div class="annotation">
                  <h1>Spinner.js</h1>
              </div>
          </li>
        
        
        
        <li id="section-1">
            <div class="annotation">
              
              <div class="pilwrap ">
                <a class="pilcrow" href="#section-1">&#182;</a>
              </div>
              <p>Spinner</p>

            </div>
            
            <div class="content"><div class='highlight'><pre>
<span class="hljs-keyword">import</span> React, { memo } <span class="hljs-keyword">from</span> <span class="hljs-string">'react'</span>;
<span class="hljs-keyword">import</span> PropTypes <span class="hljs-keyword">from</span> <span class="hljs-string">'prop-types'</span>;

<span class="hljs-keyword">import</span> classNames <span class="hljs-keyword">from</span> <span class="hljs-string">'classnames'</span>;

<span class="hljs-keyword">import</span> styles <span class="hljs-keyword">from</span> <span class="hljs-string">'./Spinner.module.scss'</span>;
<span class="hljs-keyword">import</span> <span class="hljs-string">'load-awesome/css/ball-clip-rotate.css'</span>;

<span class="hljs-keyword">export</span> <span class="hljs-function"><span class="hljs-keyword">function</span> <span class="hljs-title">Spinner</span>(<span class="hljs-params">{ spinning = true }</span>) </span>{
  <span class="hljs-keyword">const</span> spinnerClasses = classNames(
    styles.main,
    { <span class="hljs-attr">invisible</span>: !spinning },
    <span class="hljs-string">'la-ball-clip-rotate'</span>,
    <span class="hljs-string">'la-sm'</span>
  );

  <span class="hljs-keyword">return</span> (
    <span class="xml"><span class="hljs-tag">&lt;<span class="hljs-name">div</span> <span class="hljs-attr">className</span>=<span class="hljs-string">{spinnerClasses}</span> <span class="hljs-attr">data-testid</span>=<span class="hljs-string">"spinner"</span>&gt;</span>
      <span class="hljs-tag">&lt;<span class="hljs-name">div</span> /&gt;</span>
    <span class="hljs-tag">&lt;/<span class="hljs-name">div</span>&gt;</span></span>
  );
}

Spinner.propTypes = {
  <span class="hljs-attr">spinning</span>: PropTypes.bool
};

<span class="hljs-keyword">export</span> <span class="hljs-keyword">default</span> memo(Spinner);</pre></div></div>
            
        </li>
        
    </ul>
  </div>
</body>
</html>
