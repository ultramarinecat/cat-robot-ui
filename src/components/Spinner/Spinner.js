//
// Spinner
//

import React, { memo } from 'react';
import PropTypes from 'prop-types';

import classNames from 'classnames';

import styles from './Spinner.module.scss';
import 'load-awesome/css/ball-clip-rotate.css';

export function Spinner({ spinning = true }) {
  const spinnerClasses = classNames(
    styles.main,
    { invisible: !spinning },
    'la-ball-clip-rotate',
    'la-sm'
  );

  return (
    <div className={spinnerClasses} data-testid="spinner">
      <div />
    </div>
  );
}

Spinner.propTypes = {
  spinning: PropTypes.bool
};

export default memo(Spinner);
