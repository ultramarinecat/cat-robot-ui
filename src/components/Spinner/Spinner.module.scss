@charset 'utf-8';

////
//// Spinner styles
////

@import 'base.scss';

.main.main {
  color: lighten($button-text-color-disabled, 15%);
  display: inline-block;
  right: rem(-10);
  top: rem(2);
}
