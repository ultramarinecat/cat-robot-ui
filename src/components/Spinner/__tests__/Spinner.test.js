import React from 'react';
import { render } from 'react-testing-library';

import { Spinner } from '../Spinner.js';

const SPINNER = 'spinner';
const HIDDEN_CLASS = 'invisible';

describe('Spinner', () => {
  it('should display a spinner', () => {
    const { container, getByTestId } = render(<Spinner />);
    const spinner = getByTestId(SPINNER);

    expect(spinner).toBeInTheDocument();
    expect(spinner).not.toHaveClass(HIDDEN_CLASS);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not display a spinner when `spinning` is false', () => {
    const { container, getByTestId } = render(<Spinner spinning={false} />);
    const spinner = getByTestId(SPINNER);

    expect(spinner).toBeInTheDocument();
    expect(spinner).toHaveClass(HIDDEN_CLASS);

    expect(container.firstChild).toMatchSnapshot();
  });
});
