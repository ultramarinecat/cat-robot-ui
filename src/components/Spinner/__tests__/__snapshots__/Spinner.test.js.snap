// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should display a spinner 1`] = `
<div
  class="main la-ball-clip-rotate la-sm"
  data-testid="spinner"
>
  <div />
</div>
`;

exports[`Spinner should not display a spinner when \`spinning\` is false 1`] = `
<div
  class="main invisible la-ball-clip-rotate la-sm"
  data-testid="spinner"
>
  <div />
</div>
`;
